platform/linuxbsd/dbus-so_wrap.c:246:10: fatal error: 'dbus/dbus.h' file not found

### Godot version

master (5ca09b322c0dea3f71de3239c2da2a72d987f67d)

### System information

Ubuntu 22.04

### Issue description

When building, I get the following warnings:
```
Warning: fontconfig libraries not found. Disabling the system fonts support.
Warning: D-Bus development libraries not found. Disabling screensaver prevention.
Warning: Speech Dispatcher development libraries not found. Disabling Text-to-Speech support.
```
Which is fine and hasn't prevented compilation in the past, but now the build fails with:
```
Compiling platform/linuxbsd/dbus-so_wrap.c ...
platform/linuxbsd/dbus-so_wrap.c:246:10: fatal error: 'dbus/dbus.h' file not found
```
So the build system is aware that I don't have the dbus development libraries installed, but it still creates the dbus wrapper, and then it is unable to use it.

### Steps to reproduce

Ensure the D-Bus development libraries are not installed.
Try build Godot on Linux.

### Minimal reproduction project

N/A

--- a/platform/linuxbsd/detect.py
+++ b/platform/linuxbsd/detect.py
@@ -327,6 +327,7 @@ def configure(env):
             env.Append(CPPDEFINES=["DBUS_ENABLED"])
             env.ParseConfig("pkg-config dbus-1 --cflags")  # Only cflags, we dlopen the library.
         else:
+            env["dbus"] = False
             print("Warning: D-Bus development libraries not found. Disabling screensaver prevention.")
 
     if env["speechd"]:

